src/node_types.ml

```diff
@@ -0,0 +1,34 @@
+type node_exports
+type node_module = <
+  id : string ;
+  exports : node_exports ;
+  parent : node_module Js.null_undefined;
+  (* in REPL V4 it is [undefined]
+    in CLI it can be [null]
+  *)
+
+  filename : string ;
+
+  loaded : Js.boolean;
+  children : node_module array ;
+  paths : string array;
+> Js.t
+ (** WARN:
+     its path name should not be changed
+     see {!Ppx_entry}
+*)
+
+type node_require = <
+  main : node_module Js.undefined;
+
+  resolve : string -> string [@bs]
+  (* @raise exception   *)
+> Js.t
```

Comment:
Yea, pretty much. I don't think there's any examples of this in the bs std lib, but here's an example from reason-js: https://github.com/BuckleTypes/reason-js/blob/master/src/dom/nodes/nodeRe.re#L41

We've also more or less settled on camelCase rather than snake_case for for js apis, so the appropriate name would be `setId`.